servers/src/main/java/tachyon/worker/block/meta/StorageDir.java

```diff
@@ -62,6 +64,56 @@ public StorageDir(StorageTier tier, int dirIndex, long capacityBytes, String dir
     mUserIdToTempBlockIdsMap = new HashMap<Long, Set<Long>>(200);
   }
 
+  public static StorageDir newStorageDir(StorageTier tier, int dirIndex, long capacityBytes,
+      String dirPath) {
+    StorageDir dir = new StorageDir(tier, dirIndex, capacityBytes, dirPath);
+    dir.initializeMeta();
+    return dir;
+  }
+
+  /**
+   * Initialize meta data for existing blocks in this StorageDir
+   *
+   * Only paths satisfying the contract defined in {@link BlockMetaBase#commitPath()} are legal,
+   * should be in format like {dir}/{blockId}. other paths will be deleted.
+   */
+  private void initializeMeta() {
```

Comment:
One advantage of making `newStorageDir` a public static method (b.t.w. thanks for doing this in this PR!) is, it is legitimate to fail or throw exceptions for a static method (while it is tricky for constructor to fail or throw exceptions). So I think in `initializeMeta`, it is perfectly fine to throw these `IOExceptions` it sees internally to caller of `newStorageDir`, and that is the purpose. Converting those `IOExceptions` to `LOG.error` is risky because they deserve being noticed and handled (maybe by Tachyon admin)